src/components/GameClock.cpp

```diff
@@ -0,0 +1,106 @@
+#include "GameClock.hpp"
+#include <RTTI/RTTI_GameClock.hpp>
+#include <Scene/BsSceneObject.h>
+#include <Math/BsMath.h>
+
+namespace REGoth
+{
+  constexpr float SECONDS_IN_A_MINUTE = 60;
+  constexpr float SECONDS_IN_AN_HOUR  = 60 * SECONDS_IN_A_MINUTE;
+  constexpr float SECONDS_IN_A_DAY    = 24 * SECONDS_IN_AN_HOUR;
+  constexpr float CLOCK_SPEED_FACTOR  = SECONDS_IN_A_DAY/6000; // see https://forum.worldofplayers.de/forum/threads/396326-Tipp-Tageweise-springen-%28auch-zur%C3%BCck%29?p=6231841&viewfull=1#post6231841
+
+  GameClock::GameClock(const bs::HSceneObject& parent)
+    : bs::Component(parent)
+  { }
+
+  void GameClock::update()
```

Comment:
There is also `fixedUpdate` which should be better here. Otherwise we may lose some time or weird stuff happens on long frames.